packages/compartment-mapper/src/policy.js

```diff
@@ -115,60 +120,58 @@ export const detectAttenuators = policy => {
  * Verifies if a module identified by `namingKit` can be a dependency of a package per `packagePolicy`.
  * `packagePolicy` is required, when policy is not set, skipping needs to be handled by the caller.
  *
- * @param {PackageNamingKit} namingKit
+ * @param {CanonicalName} canonicalName
  * @param {PackagePolicy} packagePolicy
  * @returns {boolean}
  */
-export const dependencyAllowedByPolicy = (namingKit, packagePolicy) => {
-  if (namingKit.isEntry) {
-    // dependency on entry compartment should never be allowed
-    return false;
-  }
-  const canonicalName = generateCanonicalName(namingKit);
+export const dependencyAllowedByPolicy = (canonicalName, packagePolicy) => {
   return !!policyLookupHelper(packagePolicy, 'packages', canonicalName);
 };
 
 /**
- * Returns the policy applicable to the canonicalName of the package
- *
+ * Generates the {@link SomePackagePolicy} value to be used in {@link CompartmentDescriptor.policy}
  * @overload
- * @param {PackageNamingKit} namingKit - a key in the policy resources spec is derived from these
- * @param {SomePolicy} policy - user supplied policy
- * @returns {SomePackagePolicy} packagePolicy if policy was specified
+ * @param {CanonicalName | typeof ATTENUATORS_COMPARTMENT} label
+ * @param {object} options Options
+ * @param {SomePolicy} options.policy User-supplied policy
```

Comment:
If the intention behind this was to ensure that if the `options` arg gets provided, it has to contain `policy` at least, I don't see it working when I type `makePackagePolicy('asdf', {})`